src/DynamoCoreWpf/ViewModels/PackageManager/PackageViewModel.cs

```diff
@@ -410,6 +410,25 @@ private bool IsOwner()
             return packageManagerClient.DoesCurrentUserOwnPackage(Model, dynamoModel.AuthenticationManager.Username);
         }
 
+        private bool CanPublishNewPackage()
+        {
+            if (!CanPublish) return false;
+
+            return packageManagerClient.DoesCurrentUserOwnPackage(Model, dynamoModel.AuthenticationManager.Username) ||
+                PackageNotPublishedAndUserIsHolder(Model, dynamoModel.AuthenticationManager.Username);
+        }
+
+
+        // Utility function to assert if a local package can be published
+        // If the current user is the package holder and a package with that name has not been published yet, return true
+        private bool PackageNotPublishedAndUserIsHolder(Package package, string username)
+        {
+            bool userIsHolder = package.CopyrightHolder != null && package.CopyrightHolder.Equals(username);
```

Comment:
Does this mean that the package cannot be republished by someone who hasn't published it before? Is the `copyright` information coming from the `pkg.json` file? What if it's missing?

Reply:
I guess I'm confused between `UserIsHolder`  vs `Owner`. It looks like `DoesCurrentUserOwnPackage` checks for package maintainers while here you're checking for `CopyrightHolder`, so they're 2 different things?

Reply:
How about renaming this function as `IsPackageUnpublishedAndUserHolder`?